edb/edgeql/compiler/stmt.py

```diff
@@ -587,9 +587,8 @@ def init_stmt(
     ctx.stmt = irstmt
     if ctx.toplevel_stmt is None:
         parent_ctx.toplevel_stmt = ctx.toplevel_stmt = irstmt
-        ctx.path_scope = parent_ctx.path_scope
-    else:
-        ctx.path_scope = parent_ctx.path_scope.attach_fence()
+
+    ctx.path_scope = parent_ctx.path_scope.attach_fence()
```

Comment:
> they would end up sharing the same scope node inappropriately.

Can we have a test for that?

Reply:
Basically all function tests are the test for that with this rework, hence the change.